packages/three-vrm-core/src/expressions/VRMExpressionMaterialColorBind.ts

```diff
@@ -75,84 +96,119 @@ export class VRMExpressionMaterialColorBind implements VRMExpressionBind {
+    const alpha = this._initAlphaBindState();
+    this._state = { color, alpha };
+  }
 
-    if (propertyName == null) {
-      console.warn(
-        `Tried to add a material color bind to the material ${
-          material.name ?? '(no name)'
-        }, the type ${type} but the material or the type is not supported.`,
-      );
+  public applyWeight(weight: number): void {
+    const { color, alpha } = this._state;
 
-      this._state = null;
-    } else {
-      const target = (material as any)[propertyName] as THREE.Color;
+    if (color != null) {
+      const { propertyName, deltaValue } = color;
 
-      const initialValue = target.clone();
+      const target = (this.material as any)[propertyName] as THREE.Color;
+      if (target != undefined) {
+        target.add(_color.copy(deltaValue).multiplyScalar(weight));
+      }
+    }
 
-      // 負の値を保持するためにColor.subを使わずに差分を計算する
-      const deltaValue = new THREE.Color(
-        targetValue.r - initialValue.r,
-        targetValue.g - initialValue.g,
-        targetValue.b - initialValue.b,
-      );
+    if (alpha != null) {
+      const { propertyName, deltaValue } = alpha;
 
-      this._state = {
-        propertyName,
-        initialValue,
-        deltaValue,
-      };
+      const target = (this.material as any)[propertyName] as number;
+      if (target != undefined) {
+        ((this.material as any)[propertyName] as number) += deltaValue * weight;
+      }
     }
   }
 
-  public applyWeight(weight: number): void {
-    if (this._state == null) {
-      // warning is already emitted in constructor
-      return;
-    }
+  public clearAppliedWeight(): void {
+    const { color, alpha } = this._state;
 
-    const { propertyName, deltaValue } = this._state;
+    if (color != null) {
+      const { propertyName, initialValue } = color;
 
-    const target = (this.material as any)[propertyName] as THREE.Color;
-    if (target === undefined) {
-      return;
-    } // TODO: we should kick this at `addMaterialValue`
+      const target = (this.material as any)[propertyName] as THREE.Color;
+      if (target != undefined) {
+        target.copy(initialValue);
+      }
+    }
 
-    target.add(_color.copy(deltaValue).multiplyScalar(weight));
+    if (alpha != null) {
+      const { propertyName, initialValue } = alpha;
 
-    if (typeof (this.material as any).shouldApplyUniforms === 'boolean') {
-      (this.material as any).shouldApplyUniforms = true;
+      const target = (this.material as any)[propertyName] as number;
+      if (target != undefined) {
+        ((this.material as any)[propertyName] as number) = initialValue;
+      }
     }
   }
 
-  public clearAppliedWeight(): void {
-    if (this._state == null) {
-      // warning is already emitted in constructor
-      return;
+  private _initColorBindState(): ColorBindState | null {
+    const { material, type, targetValue } = this;
+
+    const propertyNameMap = this._getPropertyNameMap();
```

Comment:
we don't need to call `_getPropertyNameMap` twice separately in `_initColorBindState` and `_initAlphaBindState` since material will not change inside constructor? Anyway this won't be a trouble so optional.

Reply:
Yes, it's kinda verbose. How do you think if we have a private member `_propertyNameMap` to cache the result?

Reply:
mm, it's only for initialization, it doesn't affect the runtime performance. I think we're safe to leave it as is.